pkg/apb/executor.go

```diff
@@ -0,0 +1,110 @@
+package apb
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	logging "github.com/op/go-logging"
+	"github.com/openshift/ansible-service-broker/pkg/clients"
+	"github.com/pborman/uuid"
+	"k8s.io/kubernetes/pkg/api/v1"
+)
+
+type ClusterConfig struct {
+	InCluster bool
+	Target    string
+	User      string
+	Password  string `yaml:"pass"`
+}
+
+func ExecuteApb(
```

Comment:
Mentioned this to zeus, I'm picturing this function effectively turning into something much more generic that delegates the actual running of the apb to a cluster specific runner that implements an interface.